packages/web/src/components/switch/Switch.tsx

```diff
@@ -8,13 +8,19 @@ type SwitchProps = {
   isOn: boolean
   isDisabled?: boolean
   handleToggle: MouseEventHandler<Element>
+  allowCheckedWhileDisabled?: boolean
 }
 
-const Switch = ({ isOn, handleToggle, isDisabled = false }: SwitchProps) => {
+const Switch = ({
+  isOn,
+  handleToggle,
+  isDisabled = false,
+  allowCheckedWhileDisabled
+}: SwitchProps) => {
   return (
     <div className={cn(styles.container, { [styles.disabled]: isDisabled })}>
       <input
-        checked={isDisabled ? false : isOn}
+        checked={allowCheckedWhileDisabled ? isOn : isDisabled ? false : isOn}
```

Comment:
oh wow... if a checked toggle was then disabled, we used to show it as unchecked? 😵‍💫 

I guess that kinda makes sense, but really feels like it should come from the consuming component (ie `isOn` could be set to `someVar && !disabled`

Reply:
totally agree, but not wanting to change all consumers rn